pkg/controllers/workspace_controller.go

```diff
@@ -253,6 +269,30 @@ func (c *WorkspaceReconciler) updateControllerRevision(ctx context.Context, wObj
 
 	annotations[WorkspaceRevisionAnnotation] = currentHash
 	wObj.SetAnnotations(annotations)
+	if wObj.Inference != nil {
+		if previousWObj != nil && !compareAdapters(previousWObj.Inference.Adapters, wObj.Inference.Adapters) {
+			deployment := &appsv1.Deployment{}
+			if err := c.Get(ctx, types.NamespacedName{
+				Name:      wObj.Name,
+				Namespace: wObj.Namespace,
+			}, deployment); err != nil {
+				if !errors.IsNotFound(err) {
+					klog.ErrorS(err, "failed to get deployment", "deplyment", wObj.Name)
+				}
```

Comment:
You can add an annotation with revhash in the deployment, if the revhash is equal to the hash you plan to apply, you can skip the update.

Reply:
I can add it, but if the code is going to this position but still find the same revHash, it must have some errors, so I am reporting an error here